paddle/phi/infermeta/spmd_rules/layer_norm.cc

```diff
@@ -365,16 +389,17 @@ SpmdInfo LayerNormGradInferSpmd(const DistMetaTensor& x,
     x_dist_attr = std::move(dist_attrs[0]);
     mean_dist_attr = std::move(dist_attrs[1]);
     variance_dist_attr = std::move(dist_attrs[2]);
-    grad_dist_attr = std::move(dist_attrs[3]);
+    out_grad_dist_attr = std::move(dist_attrs[3]);
   } else {
     x_dist_attr = GetReplicatedDistAttr(dist_attrs[0]);
     mean_dist_attr = GetReplicatedDistAttr(dist_attrs[1]);
     variance_dist_attr = GetReplicatedDistAttr(dist_attrs[2]);
-    grad_dist_attr = GetReplicatedDistAttr(dist_attrs[3]);
+    out_grad_dist_attr = GetReplicatedDistAttr(dist_attrs[3]);
   }
   // TODO(liuzhenhai): support sharded scale and bias
   TensorDistAttr scale_dist_attr = GetReplicatedDistAttr(scale.dist_attr());
   TensorDistAttr bias_dist_attr = GetReplicatedDistAttr(bias.dist_attr());
+  TensorDistAttr x_grad_dist_attr = GetReplicatedDistAttr(out_grad_dist_attr);
```

Comment:
why replicated instead of x_grad_dist_attr = out_grad_dist_attr?

Reply:
Typo, fixed